planb/update_cluster.py

```diff
@@ -311,11 +311,32 @@ def create_instance(ec2: object, volume: dict, saved_instance: dict,
     logger.info(
         "Creating new instance with IP {}".format(params['PrivateIpAddress'])
     )
-    ec2.run_instances(**params)
+    response = ec2.run_instances(**params)
+    instance_id = response['Instances'][0]['InstanceId']
+    create_tags(ec2, volume['VolumeId'], {'planb:operation:new-instance-id': instance_id})
     set_state(ec2, volume, 'created')
 
 
-def configure_instance(ec2: object, volume: dict, options: dict):
+def pre_configure_instance(ec2: object, volume: dict, saved_instance: dict):
+    if 'PublicIpAddress' in saved_instance:
+        instance_id = tags_as_dict(volume.get('Tags', [])).get('planb:operation:new-instance-id')
+
+        instance = get_instance(ec2, instance_id)
+        if instance['State']['Name'] != 'running':
+            return
+
+        logger.info(
+            "Associating new instance with Public IP {}".format(saved_instance['PublicIpAddress'])
+        )
+        ec2.associate_address(InstanceId=instance_id, PublicIp=saved_instance['PublicIpAddress'])
+        logger.info("Rebooting new instance")
+        ec2.reboot_instances(InstanceIds=[instance_id])
```

Comment:
I wonder if it still works with create_cluster.py: there we don't reboot the instance.  Can you please describe again why is this needed?  I remember that Taupage was not able to find the EBS volume unless public IP was associated.  Maybe it is a bug in Taupage?